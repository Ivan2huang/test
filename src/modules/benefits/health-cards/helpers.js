/* eslint-disable import/prefer-default-export */
export const getQRCodeXML = ({
  fullName,
  membershipNumber,
  certificateNumber,
  insurerCode,
  insurerName,
  insurerType,
  policyNumber,
  expiryDate,
}) =>
  [
    '<?xml version="1.0"?>',
    '<medical_card>',
    '<Card_type>',
    insurerType,
    '</Card_type>',
    '<insurance_company_code>',
    insurerCode,
    '</insurance_company_code>',
    '<insurance_company_name>',
    insurerName,
    '</insurance_company_name>',
    '<insured_name>',
    fullName,
    '</insured_name>',
    '<policy_no>',
    policyNumber,
    '</policy_no>',
    '<certificate_no>',
    certificateNumber,
    '</certificate_no>',
    '<membership_no>',
    membershipNumber,
    '</membership_no>',
    '<expiry_date>',
    expiryDate,
    '</expiry_date>',
    '</medical_card>',
  ].join('');
