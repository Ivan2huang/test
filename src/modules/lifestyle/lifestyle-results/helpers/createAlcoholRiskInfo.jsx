import React from 'react';

import { AlcoholIcon, NoAlcoholIcon } from '../../../../icons';
import { formatMessage } from '../../../../helpers/helpers';

const createAlcoholRiskInfo = (intl, theme, alcoholRisk) => {
  const alcoholRiskMapping = {
    None: `${theme.lifeStyleResultOK}`,
    RegularDrinker: `${theme.lifeStyleResultWarning}`,
    HeavyDrinker: `${theme.lifeStyleResultAlarm}`,
  };

  return {
    id: 'alcohol',
    color: alcoholRiskMapping[alcoholRisk],
    CardIcon:
      alcoholRisk === 'None' ? (
        <NoAlcoholIcon />
      ) : (
        <AlcoholIcon color={alcoholRiskMapping[alcoholRisk]} />
      ),
    title: formatMessage(
      intl,
      'lifestyle.lifestyleResults.alcoholRiskCard.title',
      'Alcohol',
    ),
    subTitle: formatMessage(
      intl,
      `lifestyle.lifestyleResults.alcoholRiskCard.subTitle.${alcoholRisk}`,
      alcoholRisk,
    ),
    description: formatMessage(
      intl,
      `lifestyle.lifestyleResults.alcoholRiskCard.description.${alcoholRisk}`,
      alcoholRisk,
    ),
  };
};

export default createAlcoholRiskInfo;
