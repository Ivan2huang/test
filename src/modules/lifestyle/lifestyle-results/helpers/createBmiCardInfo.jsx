import React from 'react';

import Typography from '../../../../uiComponents/Typography';
import { formatMessage } from '../../../../helpers/helpers';

export const formatBmiScore = bmiScore => {
  const score = parseFloat(bmiScore);
  if (Number.isNaN(score)) {
    return '';
  }
  return (+bmiScore).toFixed(1);
};

const createBmiCardInfo = (intl, theme, bmiScore, bmiClass) => {
  const bmiColorMapping = {
    Healthy: `${theme.lifeStyleResultOK}`,
    Overweight: `${theme.lifeStyleResultWarning}`,
    Obese: `${theme.lifeStyleResultAlarm}`,
    Underweight: `${theme.lifeStyleResultWarning}`,
  };

  return {
    id: 'bmi',
    color: bmiColorMapping[bmiClass],
    CardIcon: (
      <Typography type="style-1" color={theme.grey3} fontWeight="bold">
        {formatBmiScore(bmiScore)}
      </Typography>
    ),
    title: formatMessage(
      intl,
      'lifestyle.lifestyleResults.bmiCard.title',
      'BMI',
    ),
    subTitle: formatMessage(
      intl,
      `lifestyle.lifestyleResults.bmiCard.subTitle.${bmiClass}`,
      bmiClass,
    ),
    description: formatMessage(
      intl,
      `lifestyle.lifestyleResults.bmiCard.description.${bmiClass}`,
      bmiClass,
    ),
  };
};

export default createBmiCardInfo;
