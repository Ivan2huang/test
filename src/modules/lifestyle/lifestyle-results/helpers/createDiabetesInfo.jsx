import React from 'react';

import { HeartIcon } from '../../../../icons';
import { formatMessage } from '../../../../helpers/helpers';

const createDiabetesInfo = (intl, theme, diabetesRisk) => {
  const alcoholRiskMapping = {
    Low: `${theme.lifeStyleResultOK}`,
    High: `${theme.lifeStyleResultWarning}`,
    VeryHigh: `${theme.lifeStyleResultAlarm}`,
  };

  return {
    id: 'diabetes',
    color: alcoholRiskMapping[diabetesRisk],
    CardIcon: <HeartIcon color={alcoholRiskMapping[diabetesRisk]} />,
    title: formatMessage(
      intl,
      'lifestyle.lifestyleResults.diabetesCard.title',
      'Diabetes',
    ),
    subTitle: formatMessage(
      intl,
      `lifestyle.lifestyleResults.diabetesCard.subTitle.${diabetesRisk}`,
      diabetesRisk,
    ),
    description: formatMessage(
      intl,
      `lifestyle.lifestyleResults.diabetesCard.description.${diabetesRisk}`,
      diabetesRisk,
    ),
  };
};

export default createDiabetesInfo;
