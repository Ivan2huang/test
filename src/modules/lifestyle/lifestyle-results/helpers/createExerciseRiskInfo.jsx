import React from 'react';

import { ExerciseIcon } from '../../../../icons';
import { formatMessage } from '../../../../helpers/helpers';

const createExerciseRiskInfo = (intl, theme, exerciseRisk) => {
  const exerciseRiskMapping = {
    High: `${theme.lifeStyleResultOK}`,
    LowToModerate: `${theme.lifeStyleResultWarning}`,
    VeryLow: `${theme.lifeStyleResultAlarm}`,
  };

  return {
    id: 'exercise',
    color: exerciseRiskMapping[exerciseRisk],
    CardIcon: <ExerciseIcon color={exerciseRiskMapping[exerciseRisk]} />,
    title: formatMessage(
      intl,
      'lifestyle.lifestyleResults.exerciseRiskCard.title',
      'Exercise',
    ),
    subTitle: formatMessage(
      intl,
      `lifestyle.lifestyleResults.exerciseRiskCard.subTitle.${exerciseRisk}`,
      exerciseRisk,
    ),
    description: formatMessage(
      intl,
      `lifestyle.lifestyleResults.exerciseRiskCard.description.${exerciseRisk}`,
      exerciseRisk,
    ),
  };
};

export default createExerciseRiskInfo;
