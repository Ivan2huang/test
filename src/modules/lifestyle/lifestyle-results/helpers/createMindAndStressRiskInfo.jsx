import React from 'react';

import { StressIcon } from '../../../../icons';
import { formatMessage } from '../../../../helpers/helpers';

const createMindAndStressRiskInfo = (intl, theme, mindAndStressRisk) => {
  const mindAndStressRiskMapping = {
    Normal: `${theme.lifeStyleResultOK}`,
    BeCareful: `${theme.lifeStyleResultWarning}`,
    AtRisk: `${theme.lifeStyleResultAlarm}`,
  };

  return {
    id: 'mindAndStress',
    color: mindAndStressRiskMapping[mindAndStressRisk],
    CardIcon: (
      <StressIcon color={mindAndStressRiskMapping[mindAndStressRisk]} />
    ),
    title: formatMessage(
      intl,
      'lifestyle.lifestyleResults.mindAndStressRiskCard.title',
      'Mind & stress',
    ),
    subTitle: formatMessage(
      intl,
      `lifestyle.lifestyleResults.mindAndStressRiskCard.subTitle.${mindAndStressRisk}`,
      mindAndStressRisk,
    ),
    description: formatMessage(
      intl,
      `lifestyle.lifestyleResults.mindAndStressRiskCard.description.${mindAndStressRisk}`,
      mindAndStressRisk,
    ),
  };
};

export default createMindAndStressRiskInfo;
