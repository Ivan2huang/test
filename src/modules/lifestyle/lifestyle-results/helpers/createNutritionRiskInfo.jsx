import React from 'react';

import { NutritionIcon } from '../../../../icons';
import { formatMessage } from '../../../../helpers/helpers';

const createNutritionRiskInfo = (intl, theme, nutritionRisk) => {
  const nutritionRiskMapping = {
    Low: `${theme.lifeStyleResultOK}`,
    Moderate: `${theme.lifeStyleResultWarning}`,
    High: `${theme.lifeStyleResultAlarm}`,
  };

  return {
    id: 'nutrition',
    color: nutritionRiskMapping[nutritionRisk],
    CardIcon: <NutritionIcon color={nutritionRiskMapping[nutritionRisk]} />,
    title: formatMessage(
      intl,
      'lifestyle.lifestyleResults.nutritionRiskCard.title',
      'Nutrition',
    ),
    subTitle: formatMessage(
      intl,
      `lifestyle.lifestyleResults.nutritionRiskCard.subTitle.${nutritionRisk}`,
      nutritionRisk,
    ),
    description: formatMessage(
      intl,
      `lifestyle.lifestyleResults.nutritionRiskCard.description.${nutritionRisk}`,
      nutritionRisk,
    ),
  };
};

export default createNutritionRiskInfo;
