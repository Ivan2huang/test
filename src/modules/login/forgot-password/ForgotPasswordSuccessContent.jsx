/* eslint-disable react/jsx-one-expression-per-line */

import React from 'react';
import * as PropTypes from 'prop-types';
import { compose } from 'redux';
import { withRouter } from 'next/router';
import { injectIntl } from 'react-intl';

import { Box, Button } from '@material-ui/core';

import { formatMessage, navigateTo } from '../../../helpers/helpers';
import paths from '../../../helpers/paths';
import ButtonGroup from '../../../uiComponents/ButtonGroup';
import Typography from '../../../uiComponents/Typography';

const ForgotPasswordSuccessContent = ({
  intl,
  router,
  forgotPassword,
  showBackBtn,
}) => {
  const { email, verify } = router.query;
  const isFirstLogin = String(verify).toLowerCase();
  return (
    <>
      <Box>
        <Typography type="style-1">
          {formatMessage(
            intl,
            'forgotPasswordSuccess.Header.title',
            'Check your email',
          )}
        </Typography>
      </Box>
      <Box mt={4}>
        <Typography type="style-6">
          {formatMessage(
            intl,
            'forgotPasswordSuccess.Header.description-yes',
            `We've sent the reset password link to ${email}. If you don't see the email, check your junk or spam folders.`,
            { email },
          )}
        </Typography>
      </Box>
      <Box mt={4}>
        <ButtonGroup>
          {showBackBtn && (
            <Button
              variant="contained"
              color="primary"
              data-testid="btn-back-to-login"
              onClick={() => navigateTo(paths.common.login)}
            >
              {formatMessage(
                intl,
                'forgotPasswordSuccess.button.backToLogin',
                'Back to log in',
              )}
            </Button>
          )}
          <Button
            variant="outlined"
            color="secondary"
            data-testid="btn-resend-link"
            onClick={() =>
              forgotPassword(email, Boolean(isFirstLogin === 'true'))
            }
          >
            {formatMessage(
              intl,
              'forgotPasswordSuccess.button.resendLink',
              'Resend Link',
            )}
          </Button>
        </ButtonGroup>
      </Box>
    </>
  );
};

ForgotPasswordSuccessContent.propTypes = {
  intl: PropTypes.shape({}).isRequired,
  router: PropTypes.shape({}).isRequired,
  forgotPassword: PropTypes.func.isRequired,
  showBackBtn: PropTypes.bool,
};

ForgotPasswordSuccessContent.defaultProps = {
  showBackBtn: true,
};

export default compose(
  withRouter,
  injectIntl,
)(ForgotPasswordSuccessContent);
