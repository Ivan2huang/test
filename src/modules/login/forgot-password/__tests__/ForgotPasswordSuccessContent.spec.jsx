/* eslint-disable react/prop-types */

import { render, fireEvent, wait } from '@testing-library/react';
import React from 'react';
import { withRouter } from 'next/router';

import ForgotPasswordSuccessContent from '../ForgotPasswordSuccessContent';
import withIntl from '../../../../i18n/withIntlProvider';
import { navigateTo } from '../../../../helpers/helpers';

jest.mock('../../../../helpers/helpers', () => ({
  navigateTo: jest.fn(),
  formatMessage: jest.fn((intl, id, defaultMessage) => defaultMessage),
}));

jest.mock('../../../../helpers/paths', () => ({
  common: {
    login: '/login',
  },
}));

jest.mock('../../../../uiComponents/Typography', () => props => (
  <div {...props}>Typography Component</div>
));

jest.mock(
  '../../../../uiComponents/ButtonGroup',
  () => ({ children, ...rest }) => (
    <div {...rest}>
      <span>Button Group Component</span>
      {children}
    </div>
  ),
);

describe('ForgotPasswordSuccessContent Component', () => {
  const props = {
    router: {
      query: {
        email: 'dummy@test.com',
        verify: true,
      },
    },
    forgotPassword: jest.fn(),
    showBackBtn: true,
  };
  const setUp = (componentProps = {}) => {
    const Component = withRouter(withIntl(ForgotPasswordSuccessContent));
    return render(<Component {...componentProps} />);
  };

  it('should match the snapshot', () => {
    const { container } = setUp(props);

    expect(container).toMatchSnapshot();
  });

  it('should match the snapshot when email is not provided', () => {
    const newProps = {
      ...props,
      router: {
        query: {
          email: 'test@email',
        },
      },
    };
    const { container } = setUp(newProps);

    expect(container).toMatchSnapshot();
  });

  it('should match the snapshot when not show back to login provided', () => {
    const newProps = {
      ...props,
      showBackBtn: false,
    };
    const { container } = setUp(newProps);

    expect(container).toMatchSnapshot();
  });

  it('should return to login page when click on Back button', async () => {
    const { getByTestId } = setUp(props);

    const BackButton = getByTestId('btn-back-to-login');
    fireEvent.click(BackButton);

    await wait(() => {
      expect(navigateTo).toHaveBeenCalledTimes(1);
      expect(navigateTo).toHaveBeenCalledWith('/login');
    });
  });

  it('should call forgot password api when clicked on resend link button', async () => {
    const { getByTestId } = setUp(props);

    const ResendButton = getByTestId('btn-resend-link');
    fireEvent.click(ResendButton);

    await wait(() => {
      expect(props.forgotPassword).toHaveBeenCalledTimes(1);
      expect(props.forgotPassword).toHaveBeenCalledWith('dummy@test.com', true);
    });
  });
});
