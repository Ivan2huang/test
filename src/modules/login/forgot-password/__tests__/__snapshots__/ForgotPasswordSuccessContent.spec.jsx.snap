// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPasswordSuccessContent Component should match the snapshot 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-2"
  >
    <div
      type="style-1"
    >
      Typography Component
    </div>
  </div>
  <div
    class="MuiBox-root MuiBox-root-3"
  >
    <div
      type="style-6"
    >
      Typography Component
    </div>
  </div>
  <div
    class="MuiBox-root MuiBox-root-4"
  >
    <div>
      <span>
        Button Group Component
      </span>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        data-testid="btn-back-to-login"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Back to log in
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary"
        data-testid="btn-resend-link"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Resend Link
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`ForgotPasswordSuccessContent Component should match the snapshot when email is not provided 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-47"
  >
    <div
      type="style-1"
    >
      Typography Component
    </div>
  </div>
  <div
    class="MuiBox-root MuiBox-root-48"
  >
    <div
      type="style-6"
    >
      Typography Component
    </div>
  </div>
  <div
    class="MuiBox-root MuiBox-root-49"
  >
    <div>
      <span>
        Button Group Component
      </span>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        data-testid="btn-back-to-login"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Back to log in
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary"
        data-testid="btn-resend-link"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Resend Link
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`ForgotPasswordSuccessContent Component should match the snapshot when not show back to login provided 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-92"
  >
    <div
      type="style-1"
    >
      Typography Component
    </div>
  </div>
  <div
    class="MuiBox-root MuiBox-root-93"
  >
    <div
      type="style-6"
    >
      Typography Component
    </div>
  </div>
  <div
    class="MuiBox-root MuiBox-root-94"
  >
    <div>
      <span>
        Button Group Component
      </span>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary"
        data-testid="btn-resend-link"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Resend Link
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
